### PR TITLE
[13.0][IMP] sale_payment_sheet_financial_risk: Reduce each risk field depending…

### PR DESCRIPTION
**Pull request by carlosdauden**
… of payment sheet invoice status (type of debt)

@Tecnativa TT37610

### CODE REVIEW

File: sale_payment_sheet_financial_risk/models/res_partner.py [hunk 1]
```diff
@@ -21,15 +23,20 @@ class ResPartner(models.Model):
...
     )
     sale_payment_sheet_ids = fields.One2many(
         comodel_name="sale.payment.sheet.line", inverse_name="partner_id"
     )
 
     def _get_risk_sale_payment_sheet_domain(self):
-        # When p is NewId object instance bool(p.id) is False
         commercial_partners = self.filtered(
-            lambda p: (p.id and p == p.commercial_partner_id)
+            lambda p: (p.ids == p.commercial_partner_id.ids)
```
Comment (pedrobaeza) [inline sale_payment_sheet_financial_risk/models/res_partner.py:40]: The order is assured to be the same always? If not, do:

```suggestion
            lambda p: (set(p.ids) == set(p.commercial_partner_id.ids))
```

Comment (carlosdauden) [reply sale_payment_sheet_financial_risk/models/res_partner.py:40]: p in lambda filtered is always only 1 record

Comment (carlosdauden) [reply sale_payment_sheet_financial_risk/models/res_partner.py:40]: And commercial_partner_id is a Many2One field that only contains 1 record

Comment (pedrobaeza) [reply sale_payment_sheet_financial_risk/models/res_partner.py:40]: Then why not using `p == p.commercial_partner_id`?

Comment (carlosdauden) [reply sale_payment_sheet_financial_risk/models/res_partner.py:40]: When p is NewId object instance is not equal to "Normal" registry

Modified explain message:
https://github.com/OCA/credit-control/pull/204/files#diff-9407b845adfe69cef43bd1306655f899ac7ac40987e58638425c2790365ae459R38

Comment (pedrobaeza) [reply sale_payment_sheet_financial_risk/models/res_partner.py:40]: OK then

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**